c7n/resources/opensearch.py

```diff
@@ -19,7 +20,26 @@ class resource_type(TypeInfo):
         cfn_type = 'AWS::OpenSearchServerless::Collection'
         arn = "arn"
         permission_prefix = 'aoss'
-        augment = universal_augment
+
+    def augment(self, resources):
+        client = local_session(self.session_factory).client('opensearchserverless')
+        def _augment(r):
+            tags = self.retry(client.list_tags_for_resource,
+                resourceArn=r['arn'])['tags']
+            r['Tags'] = [{'Key': t['key'], 'Value': t['value']} for t in tags]
+            return r
+        resources = super().augment(resources)
+        if resources:
+          ids = [r["id"] for r in resources]
+          resources = self.retry(
+            client.batch_get_collection, ids=ids).get('collectionDetails')
```

Comment:
This should probably go into detail_spec

Reply:
This is a little different. Normally, detail_spec will specify api that accepts 1 id and return details for that id. In this case, batch_get_collection will accept a list of ids and return collection details in only 1 api call. This seems to be a lot more efficient in this particular case.

Reply:
Reading the docs, it felt like ids/names were not a required parameter. Usually they mark attributes as Required in their docs if they are so. https://boto3.amazonaws.com/v1/documentation/api/latest/reference/services/opensearchserverless/client/batch_get_collection.html
But here, it seems like either id or name is a necessary input for this api to work

Reply:
yeah, I will update the code.

Reply:
code updated. Thanks.